### PR TITLE
test(watch): start adding tests

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: tests/playwright-test/playwright-test-fixtures.ts [hunk 1]
```diff
@@ -246,28 +267,43 @@ type RunOptions = {
...
+    .extend<CommonFixtures, CommonWorkerFixtures>(commonFixtures)
     .extend<ServerFixtures, ServerWorkerOptions>(serverFixtures)
     .extend<Fixtures>({
       writeFiles: async ({}, use, testInfo) => {
         await use(files => writeFiles(testInfo, files));
       },
 
       runInlineTest: async ({ childProcess }, use, testInfo: TestInfo) => {
-        await use(async (files: Files, params: Params = {}, env: Env = {}, options: RunOptions = {}, beforeRunPlaywrightTest?: ({ baseDir: string }) => Promise<void>) => {
+        const cacheDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'playwright-test-cache-'));
```
Comment (dgozman) [inline tests/playwright-test/playwright-test-fixtures.ts:287]: Why not `testInfo.outputPath('cache')`?

File: tests/playwright-test/watch.spec.ts [hunk 2]
```diff
@@ -89,3 +89,81 @@ test('should print dependencies in ESM mode', async ({ runInlineTest, nodeVersio
...
+    `,
+  }, {});
+  await testProcess.waitForOutput('a.test.ts:5:11 › passes');
+  await testProcess.waitForOutput('Waiting for file changes.');
+});
+
+test('should quit on Q', async ({ runWatchTest }) => {
+  const testProcess = await runWatchTest({}, {});
+  await testProcess.waitForOutput('Waiting for file changes.');
+  testProcess.process.stdin!.write('q');
```
Comment (dgozman) [inline tests/playwright-test/watch.spec.ts:106]: Perhaps introduce a helper `TestChildProcess.write()`?

**Review by dgozman — APPROVED** [review]

File: tests/playwright-test/playwright-test-fixtures.ts (hunk 1 above)
Comment (pavelfeldman) [reply tests/playwright-test/playwright-test-fixtures.ts:287]: It'll clash with the test workspace.

File: tests/playwright-test/watch.spec.ts (hunk 2 above)
Comment (pavelfeldman) [reply tests/playwright-test/watch.spec.ts:106]: I'll follow up if we get many

**Pull request merged**